### DISCUSSION TITLE
[next/image] Support for AVIF image format

### DISCUSSION THREAD

**Discussion by abriginets** [vercel/next.js#27882] [Help] [answered]
### Describe the feature you'd like to request

Hello. Is that possible to add support for AVIF image format for next/image module at this moment? If you're not familiar with it here's a good topic to learn from [Comparing AVIF vs WebP file sizes at the same DSSIM](https://www.ctrl.blog/entry/webp-avif-comparison.html) (TLDR; AVIF-encoded images are ~20% lighter than WebP in most of the cases) Many Chrome and Opera versions supports it, and also latest Firefox released major version with AVIF support. Safari doesn't support it yet but considering their latest experience with WebP - they will at some point.

### Describe the solution you'd like

NextJS should fetch support table from [caniuse.com](https://caniuse.com/?search=avif) and compare it with visitor's User-Agent and see if browser version supports AVIF. This way NextJS should serve AVIFs to only those visotors whose browser does support it and fallback to WebP otherwise.

### Describe alternatives you've considered

\-

**Comment by rahul3103** [comment]
``` 
    <picture>
       <source srcSet={avif} type="image/avif" />
        <source srcSet={webp} type="image/webp" />
        <source srcSet={jpg} type={`image/jpg`} />
        <StyledImage
          src={imgUrl}
          alt={alt}
          decoding="async"
          loading="lazy"
        />
     </picture>

**Comment by leerob** [comment, marked as answer]
This is available in Next.js 12 https://github.com/vercel/next.js/pull/29683
